File: src/System/AdvisorRegistrationSystem.java
package System;

import java.io.FileNotFoundException;
import java.util.Optional;
import java.util.Scanner;

import Enums.FilterType;
import data.DataManager;
import person.Advisor;

//Kaan Camci 150119063
public class AdvisorRegistrationSystem {

	private Scanner scanner;
	private RegistrationSystem registrationSystem1;

	public AdvisorRegistrationSystem(RegistrationSystem registrationSystem) throws FileNotFoundException {
		scanner = new Scanner(System.in);
		registrationSystem1 = registrationSystem;
		AdvisorLogin();
	}

	private void AdvisorLogin() throws FileNotFoundException {
		Advisor currentUser = null;

		System.out.println("Please provide your ID:");
		System.out.println("----\nSuggestion: Enter \"????\"");
		while (true) {
			String providedID = scanner.nextLine();
			Optional<Advisor> currentOptionalAdvisor = DataManager.getInstance().findAdvisor(providedID, FilterType.ID);
			if (currentOptionalAdvisor.isPresent()) {
				currentUser = currentOptionalAdvisor.get();
				advisorMenu(currentUser);
				break;
			} else {
				System.out.print("Advisor not found, please try again: ");
			}
		}

	}

	private void advisorMenu(Advisor currentUser) throws FileNotFoundException {

		boolean validInput = false;

		int menuChoice = 0;

		while (menuChoice != 3) {

			System.out.println("Please choose a menu: ");
			System.out.println("1-Registration Applications");
			System.out.println("2-Sign Out");

			validInput = false;

			while (!validInput) {

				menuChoice = scanner.nextInt();

				switch (menuChoice) {
				case 1:
					validInput = true;
					showApplications(currentUser);
					break;
				case 2:
					validInput = true;
					signOut();
					break;
				default:
					System.out.print("The input is not valid, please provide a valid input.");
				}
			}
		}

	}

	private void showApplications(Advisor currentUser) {

		currentUser.showApplications();

	}

	private void signOut() throws FileNotFoundException {
		registrationSystem1.menu();
	}

}

File: pythonCode/AdvisorRegistrationSystem.py
from FilterType import FilterType
from DataManager import DataManager
from Advisor import Advisor
from Logger import Logger

class AdvisorRegistrationSystem():

    #Holding registrationSystem object for returning after signing out#

    def __init__(self, registrationSystem):
        #Constructor of AdvisorRegistrationSystem

        self.__log = Logger.getLogger("logs")

        self.__registrationSystem1 = registrationSystem
        self.advisorLogin()

    def advisorLogin(self):

        #Searching for a Advisor with a given id. If there is an id equal with providedId currentUser is that Advisor.#
        currentUser = None
        self.__log.info("Please provide an Advisor ID:")
        self.__log.info("----\nSuggestion: Enter \"150097\"")#
        while True:
            providedID = input()
            currentUser = DataManager.getInstance().findAdvisor(providedID, FilterType.ID)
            self.advisorMenu(currentUser)

    def advisorMenu(self, currentUser):
        #Showing Advisor's menu choices and making a choice#
        validInput = False
        menuChoice = str(0)
        while menuChoice != str(2):
            self.__log.info("Choose a menu: ")
            self.__log.info("1-Simulation Results")
            self.__log.info("2-Exit")

            validInput = False
            while not validInput:
                menuChoice = input()
                if menuChoice==str(1):
                    validInput = True
                    #simulasyon sonuçları
                elif menuChoice==str(2):
                    validInput = True
                    self.signOut()
                else:
                    self.__log.info("The input is not valid, please provide a valid input.")

    def signOut(self):
        #returning back to RegistrationSystem's menu#
        self.__registrationSystem1.menu()
